test/framework/src/main/java/org/elasticsearch/logsdb/datageneration/MappingTemplate.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.logsdb.datageneration;
+
+import java.util.Map;
+
+public record MappingTemplate(Map<String, Entry> mapping) {
```

Comment:
Nit: let's add comment classes everywhere, so it's easier for someone to understand what functionality is provided.